### PR TITLE
Implement Reddit provider

### PR DESCRIPTION
**Pull request by stnrd**
This MR contains the implementation of the Reddit login provider. 

I have created this MR after a discussion topic I have created 2 weeks ago. 

As mentioned in there is that Reddit only gives sessions for 1 hour and doesn't share email addresses. 

Discussion: https://github.com/pocketbase/pocketbase/discussions/1851

Just let me know if I need to change something.

### CODE REVIEW

File: tools/auth/reddit.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+	rawUser := map[string]any{}
+	if err := json.Unmarshal(data, &rawUser); err != nil {
+		return nil, err
+	}
+
+	fmt.Println(string(data))
+
+	extracted := struct {
+		Login            string `json:"login"`
```
Comment (ganigeorgiev) [inline tools/auth/reddit.go:45]: What is `Login`?

File: tools/auth/reddit.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+func NewRedditProvider() *Reddit {
+	return &Reddit{&baseProvider{
+		scopes:     []string{"identity"},
+		authUrl:    "https://www.reddit.com/api/v1/authorize",
+		tokenUrl:   "https://www.reddit.com/api/v1/access_token",
+		userApiUrl: "https://oauth.reddit.com/api/v1/me",
+	}}
+}
+
+// FetchAuthUser returns an AuthUser instance based the Reddit's user api.
```
Comment (ganigeorgiev) [inline tools/auth/reddit.go:30]: If available, could you also add a link to the official user API documentation?

File: tools/auth/reddit.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	rawUser := map[string]any{}
+	if err := json.Unmarshal(data, &rawUser); err != nil {
+		return nil, err
+	}
+
+	fmt.Println(string(data))
```
Comment (ganigeorgiev) [inline tools/auth/reddit.go:42]: The debug statement can be removed if not needed anymore.

File: tools/auth/reddit.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+	extracted := struct {
+		Login            string `json:"login"`
+		Id               string `json:"id"`
+		Name             string `json:"name"`
+		AvatarUrl        string `json:"snoovatar_img"`
+		HasVerifiedEmail bool   `json:"has_verified_email"`
```
Comment (ganigeorgiev) [inline tools/auth/reddit.go:49]: We can remove this field if Reddit doesn't return an email.

File: tools/auth/reddit.go [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+		return nil, err
+	}
+
+	user := &AuthUser{
+		Id:          extracted.Id,
+		Name:        extracted.Name, // Reddit isn't sharing the display name only username
+		Username:    extracted.Name,
+		AvatarUrl:   extracted.AvatarUrl,
+		RawUser:     rawUser,
+		AccessToken: token.AccessToken,
```
Comment (ganigeorgiev) [inline tools/auth/reddit.go:62]: Add also `RefreshToken: token.RefreshToken,`

File: tools/auth/reddit.go [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+	}
+
+	if !extracted.HasVerifiedEmail || extracted.IsSuspended {
+		return nil, fmt.Errorf("email not verified or account suspended")
+	}
```
Comment (ganigeorgiev) [inline tools/auth/reddit.go:67]: Let's return an error only if the account is suspended, since we don't care about the email in this case.

**Review by ganigeorgiev — CHANGES_REQUESTED** [review]
Thanks! I left some minor notes, feel free to check them when you have time.

I'll review and test the provider in more details sometime later this week.

**Comment by denizkenan** [comment]
is development on this paused? Should I pick it up?

**Comment by ganigeorgiev** [comment]
I'm closing this PR since it seems abandoned and it is behind the latest changes.

If someone else want to works on this, feel free to file a new PR against the `develop` branch (ideally addressing the above comments).

**Pull request closed without merging**